server/src/com/cloud/api/response/ApiResponseSerializer.java

```diff
@@ -127,53 +151,74 @@ public static String toJSONSerializedString(ResponseObject result) {
+        if (result != null && log != null) {
+            StringBuilder sb = new StringBuilder();
+            sb.append("<?xml version=\"1.0\" encoding=\"UTF-8\"?>");
+            sb.append("<").append(result.getResponseName()).append(" cloud-stack-version=\"").append(ApiDBUtils.getVersion()).append("\">");
+            log.append("<?xml version=\"1.0\" encoding=\"UTF-8\"?>");
+            log.append("<").append(result.getResponseName()).append(" cloud-stack-version=\"").append(ApiDBUtils.getVersion()).append("\">");
 
-        if (result instanceof ListResponse) {
-            Integer count = ((ListResponse)result).getCount();
+            if (result instanceof ListResponse) {
+                Integer count = ((ListResponse)result).getCount();
 
-            if (count != null && count != 0) {
-                sb.append("<").append(ApiConstants.COUNT).append(">").append(((ListResponse)result).getCount()).append("</").append(ApiConstants.COUNT).append(">");
-            }
-            List<? extends ResponseObject> responses = ((ListResponse)result).getResponses();
-            if ((responses != null) && !responses.isEmpty()) {
-                for (ResponseObject obj : responses) {
-                    serializeResponseObjXML(sb, obj);
+                if (count != null && count != 0) {
+                    sb.append("<").append(ApiConstants.COUNT).append(">").append(((ListResponse)result).getCount()).append("</").append(ApiConstants.COUNT).append(">");
+                    log.append("<").append(ApiConstants.COUNT).append(">").append(((ListResponse)result).getCount()).append("</").append(ApiConstants.COUNT).append(">");
+                }
+                List<? extends ResponseObject> responses = ((ListResponse)result).getResponses();
+                if ((responses != null) && !responses.isEmpty()) {
+                    for (ResponseObject obj : responses) {
+                        serializeResponseObjXML(sb, log, obj);
+                    }
                 }
-            }
-        } else {
-            if (result instanceof CreateCmdResponse || result instanceof AsyncJobResponse || result instanceof AuthenticationCmdResponse) {
-                serializeResponseObjFieldsXML(sb, result);
             } else {
-                serializeResponseObjXML(sb, result);
+                if (result instanceof CreateCmdResponse || result instanceof AsyncJobResponse || result instanceof AuthenticationCmdResponse) {
+                    serializeResponseObjFieldsXML(sb, log, result);
+                } else {
+                    serializeResponseObjXML(sb, log, result);
+                }
             }
-        }
 
-        sb.append("</").append(result.getResponseName()).append(">");
-        return sb.toString();
+            sb.append("</").append(result.getResponseName()).append(">");
+            log.append("</").append(result.getResponseName()).append(">");
```

Comment:
@koushik-das I see this happening in a few places. I'm fine with it but from a performance point of view it might not be optimal.